types/gatherer.d.ts

```diff
@@ -55,28 +58,52 @@ declare global {
       trace?: Trace;
     }
 
-    export type PhaseResultNonPromise = void|LH.GathererArtifacts[keyof LH.GathererArtifacts]
+    export type PhaseArtifact = LH.GathererArtifacts[keyof LH.GathererArtifacts]
+    export type PhaseResultNonPromise = void|PhaseArtifact
     export type PhaseResult = PhaseResultNonPromise | Promise<PhaseResultNonPromise>
 
     export type GatherMode = 'snapshot'|'timespan'|'navigation';
 
-    export interface GathererMeta {
+    export type DefaultDependenciesKey = '__none__'
+    export type DependencyKey = keyof GathererArtifacts | DefaultDependenciesKey
+
+    interface GathererMetaNoDependencies {
+      /**
+       * Used to validate the dependency requirements of gatherers.
+       * If this property is not defined, this gatherer cannot be the dependency of another. */
+      symbol?: Symbol;
+      /** Lists the modes in which this gatherer can run. */
       supportedModes: Array<GatherMode>;
     }
 
+    interface GathererMetaWithDependencies<
+      TDependencies extends DependencyKey = DefaultDependenciesKey
+    > extends GathererMetaNoDependencies {
+      /**
+       * The set of required dependencies that this gatherer needs before it can compute its results.
+       * If
```

Comment:
I started to write about `'*'` purpose but I don't think we'll need it for the foreseeable future, so that'll go away.